lightly/api/api_workflow_predictions.py

```diff
@@ -0,0 +1,155 @@
+          >>>     ],
+          >>> )
+          >>> client.create_or_update_prediction_task_schema(
+          >>>     schema=schema, prediction_uuid_timestamp=timestamp
+          >>> )
+
+
+        """
+        self._predictions_api.create_or_update_prediction_task_schema_by_dataset_id(
+            body=schema,
+            dataset_id=self.dataset_id,
+            prediction_uuid_timestamp=prediction_uuid_timestamp,
+        )
+
+
+    def create_or_update_predictions(
+        self,
+        filename_to_prediction_singletons: Union[
+            Dict[str, List[PredictionSingletonClassificationRepr]],
+            Dict[str, List[PredictionSingletonObjectDetection]],
+            Dict[str, List[PredictionSingletonInstanceSegmentation]],
+            Dict[str, List[PredictionSingletonKeypointDetection]],
+        ],
+        prediction_uuid_timestamp: int,
+        progress_bar: Optional[tqdm.tqdm] = None,
+        max_workers: int = 8
+    ) -> None:
+        """Creates or updates the predictions for specific samples
+
+        Args:
+            filename_to_prediction_singletons
+                A mapping from the filename of the sample to its corresponding prediction singletons.
+                All singletons must be of the same type.
+
+            prediction_uuid_timestamp:
+                The uuid timestamp in nanoseconds since 1970. Get it via 'int(time.time())'.
+
+            progress_bar:
+                Tqdm progress bar to show how many prediction files have already been
+                uploaded.
+
+            max_workers:
+                Maximum number of workers uploading predictions in parallel.
+
+        Example:
+          >>> import time
+          >>> from tqdm import tqdm
+          >>> from lightly.api import ApiWorkflowClient
+          >>> from lightly.openapi_generated.swagger_client import (
+          >>>     PredictionTaskSchema,
+          >>>     TaskType,
+          >>>     PredictionTaskSchemaCategory,
+          >>> )
+          >>> from lightly.api.prediction_singletons import PredictionSingletonClassificationRepr
+          >>>
+          >>> client = ApiWorkflowClient(
+          >>>     token="MY_LIGHTLY_TOKEN", dataset_id="MY_DATASET_ID"
+          >>> )
+          >>>
+          >>> timestamp = int(time.time())
+          >>> filenames = client.get_filenames()
+          >>> filename_to_prediction_singletons_dummy = {
+          >>>     filename: [PredictionSingletonClassificationRepr(taskName="my-task", categoryId=i%4, score=0.9, probabilities=[0.1, 0.2, 0.3, 0.4])]
+          >>>     for i, filename in enumerate(filenames)
+          >>> }
+          >>> client.create_or_update_predictions(
+          >>>     filename_to_prediction_singletons_dummy,
+          >>>     prediction_uuid_timestamp=timestamp,
+          >>>     progress_bar=tqdm(desc="Uploading predictions", total=len(filenames), unit=" predictions")
+          >>> )
+
+
+        """
+
+        samples = retry(
```

Comment:
I see as this is a pip function and thus it makes sense to check this for sanity reasons. However I am not a big fan of querying the server again when we should already have this info. Maybe we can allow to skip this (when using from lightlyworker) somehow? Or we pass the dictionary instead of `filename<->singleton[]` as `sampleId<->singleton[]` instead?

Reply:
Probably @guarin is better to answer because he knows what data we have available in worker. Jeremy is on point, the worker should already know a sampleId. (I have not heard of sampleId though before, do we use it in worker?)